packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1403,6 +1422,18 @@ function updateAccessOfModel(context: TCGCContext): void {
   }
 }
 
+function updateSpreadModelUsageAndAccess(context: TCGCContext): void {
+  for (const sdkType of context.spreadModels?.values() ?? []) {
+    updateUsageOfModel(context, UsageFlags.Spread, sdkType);
+  }
+  for (const sdkType of context.modelsMap?.values() ?? []) {
+    // if a type only has spread usage, then it could be internal
+    if (sdkType.usage === UsageFlags.Spread) {
+      sdkType.access = "internal";
```

Comment:
If a model is only used for spread, does that mean the TCGC would not return that model in getAllModels by default?

Reply:
it will return it but with access internal